db/migrate/20171026114327_add_deleted_on_to_container_quota_and_items.rb

```diff
@@ -0,0 +1,10 @@
+class AddDeletedOnToContainerQuotaAndItems < ActiveRecord::Migration[5.0]
+  def change
+    add_column :container_quotas, :deleted_on, :datetime
+    add_index :container_quotas, :deleted_on
+
+    add_column :container_quota_items, :created_on, :datetime
```

Comment:
This will be a problem because `created_on` is the old name for the built-in ActiveRecord timestamps, and ActiveRecord still supports that name, thus ActiveRecord will auto-fill this column with Time.now when you create the record.

Reply:
That's what I had in mind, the standard created_on.  most containers tables (including container_quotas) already have this, items didn't...
Should I prefer `created_at`?

Do you see reason a custom named, manually filled column instead?  I just tried and I can override `created_on` if I'll really need.

Reply:
(the reason to want both created & deleted is we might want to answer "what were the quotas at moment T". I imagine we'd do something like `created_on <= T AND T < deleted_on`...)

Reply:
So, if you want to store the "native" creation date, you probably want a different column that created_at/created_on as those are reserved as "the date of when the row in the database was created".  If the other containers tables are inserting the "native" creation data into that column, then it's technically wrong.  We really should not override built-in ActiveRecord things, as it only leads to confusion later.

If you still want to create a created_at column, prefer using the built-in .timestamps method in migrations over adding the column manually

```ruby
change_table :container_quota_items do |t|
  t.timestamps
end

Reply:
Ah, I see what you mean.
AFAIK we don't currently override any AR timestamps.
We do get some "native" dates from k8s and stuff those into our own columns like `container_quotas.ems_created_on`, `containers.finished_at` etc. :heavy_check_mark: 

"created_on/deleted_on" in other archived container entities already means "creation/archival *persisted* at" (creation set by AR magic ts / `deleted_on = Time.now`), and we were fine with that.
Here for quotas we'll also abuse "deleted/created_on" to mean "persisted change at" and not just "persisted creation/deletion".  But all 3 are approx refresh timestamps...

In theory at some point we might prefer "collected on" timestamps as closer to truth (esp. with watch worker — @Ladas does watch give us real changed-in-k8s timestamp?), but that'd be a global move, would need our own columns on many tables, and out of scope here.

- However, yes, I do feel a temptation to achieve "subversion-like" history where several items changed in same refresh get *exactly* same timestamp, so as you do "svn log" you'll see an "atomic change".
  I'm trying to resist it :-)
  More seriously, *if* we'll want this, I think the sane design is not syncronized timestamps but archive only `container_quota`, and attach brand new items & scopes under it (items & scopes deleted_on would be unused in this scenario).

Need to head home, I'll rewrite to `t.timestamps`.